runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/Outcome.kt

```diff
@@ -39,3 +41,12 @@ public fun <T> Outcome<T>.getOrThrow(): T = when (this) {
     is Outcome.Response -> response
     is Outcome.Exception -> throw exception
 }
+
+/**
+ * Convert an outcome to a [Result]
+ */
+@InternalApi
+public fun <T> Outcome<T>.toResult(): Result<T> = when (this) {
```

Comment:
**Comment**: While we created this specifically for retry middleware, it seems generally useful and I can't think of a reason it shouldn't be `public` (non-`@InternalApi`).